src/Codebase/Application/Dto/CodebaseRequestDto.php

```diff
@@ -20,65 +20,81 @@ class CodebaseRequestDto
     public const CORE_TYPE = 'core';
 
     /**
-     * @var array<string>
+     * @var string
+     */
+    protected string $srcDirectory;
+
+    /**
+     * @var string
      */
-    protected array $projectPath;
+    protected string $toolingConfigurationPath;
 
     /**
      * @var array<string>
      */
-    protected array $corePath;
+    protected array $corePaths;
 
     /**
      * @var array<string>
      */
     protected array $coreNamespaces;
 
     /**
-     * @var string
+     * @var array<string>
      */
-    protected string $projectPrefix;
+    protected array $excludeList;
 
     /**
      * @var array<string>
      */
-    protected array $excludeList;
+    protected array $projectPaths;
+
+    /**
+     * @var array<string>
+     */
+    protected array $projectPrefixes;
 
     /**
-     * @param array<string> $projectPath
-     * @param array<string> $corePath
+     * @param string $srcDirectory
+     * @param string $toolingConfigurationPath
+     * @param array<string> $corePaths
      * @param array<string> $coreNamespaces
-     * @param string $projectPrefix
      * @param array<string> $excludeList
      */
     public function __construct(
-        array $projectPath = [],
-        array $corePath = [],
+        string $srcDirectory = '',
+        string $toolingConfigurationPath = '',
+        array $corePaths = [],
         array $coreNamespaces = [],
-        string $projectPrefix = '',
         array $excludeList = []
     ) {
-        $this->projectPath = $projectPath;
-        $this->corePath = $corePath;
+        $this->srcDirectory = $srcDirectory;
+        $this->toolingConfigurationPath = $toolingConfigurationPath;
+        $this->corePaths = $corePaths;
         $this->coreNamespaces = $coreNamespaces;
-        $this->projectPrefix = $projectPrefix;
         $this->excludeList = $excludeList;
+        $this->projectPaths = [];
+        $this->projectPrefixes = [];
     }
 
     /**
      * @return array<string>
      */
-    public function getProjectPath(): array
+    public function getProjectPaths(): array
```

Comment:
Do you mean source directories?

Reply:
directories such as `src/Pyz`, `src/CutomDir`...